password_enhancements.module

```diff
@@ -0,0 +1,416 @@
+<?php
+
+/**
+ * @file
+ * Password related enchantments.
+ */
+
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\password_enhancements\Form\Settings;
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_theme().
+ */
+function password_enhancements_theme($existing, $type, $theme, $path): array {
+  return [
+    'password_enhancements_error_messages' => [
+      'variables' => [
+        'error_message' => '',
+        'messages' => [],
+      ],
+    ],
+    'password_enhancements_policy_constraints' => [
+      'variables' => [
+        'constraints' => [],
+        'descriptions' => [],
+        'hidden' => FALSE,
+      ],
+    ],
+    'password_enhancements_logged_out_message' => [
+      'variables' => [
+        'message' => '',
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function password_enhancements_form_user_form_alter(&$form, FormStateInterface $form_state, string $form_id): void {
+  password_enhancements_form_user_register_form_alter($form, $form_state, $form_id);
+  $form['password_change_required']['#access'] = Drupal::currentUser()->hasPermission('administer user password settings');
+  array_unshift($form['actions']['submit']['#submit'], 'password_enhancements_form_user_form_alter_submit');
```

Comment:
Should this submit handler **always** run first? Because in that case, the module's weight should be changed as well otherwise other modules that are running later can prepend submit handlers before this one.

Reply:
Yes, it always has to run first, well at least before saving the entity itself, so it is not necessary to go before all of the modules's submit callback, just before the user entity save.

Reply:
Ok, this part is removed now, I'm updating the password changed date in `hook_user_presave()`.